lazy-either.js

```diff
@@ -1,46 +1,52 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
-
-const LazyEither = require('lazy-either');
-const R = require('ramda');
-const S = require('sanctuary');
-
-
-//    join :: String -> String -> String
-const join = R.curryN(2, path.join);
+const fs            = require('fs');
+
+const LazyEither    = require('lazy-either');
+const S             = require('sanctuary');
+
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
+
+
+//    Array$traverse :: Applicative f => TypeRep f -> (a -> f b) -> Array a -> f (Array b)
+//
+//    We cannot use S.traverse because LazyEither does not currently support
+//    recent versions of the Fantasy Land specification. It's thus necessary
+//    to dispatch to `map` and `ap` (rather than their prefixed equivalents).
```

Comment:
It's implementing the fantasy land spec, what are the fantasy-land prefixed methods?

It's not actively developed but it's lazily maintained (accepting PRs). Avaq's fluture lib handles the basic cases for using Either and Maybe with futures so I recommend that to people. Due to LazyEither's restrictions, it does have some interesting properties that fluture can't have because it's more general, but I'm more interested in purescript right now.